http3/http_stream.go

```diff
@@ -47,21 +47,29 @@ type stream struct {
 	buf []byte // used as a temporary buffer when writing the HTTP/3 frame headers
 
 	bytesRemainingInFrame uint64
+	seenTrailerFrame      bool
 
 	datagrams *datagrammer
+
+	onTrailersFrame func(*headersFrame) (int, error)
 }
 
 var _ Stream = &stream{}
 
 func newStream(str quic.Stream, conn *connection, datagrams *datagrammer) *stream {
 	return &stream{
-		Stream:    str,
-		conn:      conn,
-		buf:       make([]byte, 16),
-		datagrams: datagrams,
+		Stream:          str,
+		conn:            conn,
+		buf:             make([]byte, 16),
+		datagrams:       datagrams,
+		onTrailersFrame: func(*headersFrame) (int, error) { return 0, nil },
 	}
 }
 
+func (s *stream) setOnTrailersFrame(onTrailersFrame func(*headersFrame) (int, error)) {
```

Comment:
I'm a bit confused by this logic here. Why do we need to set a callback here? Why can't we just have a `requestStream.ReadTrailer() (*http.Response, error)`?

Reply:
Hm.. The reason for this is to be able to re-use the `stream` type for clients. It's my understanding that when clients read the body, `stream. Read()` is called consecutively until the body is completely exhausted. The `stream.Read()` function doesn't really know when it will yield a headers frame (which contains the trailers): Therefore, I figured that there's a few options:
1. This weird callback behavior that I have here
2. Breaking the Read() interface and potentially returning the trailers
3. Directly add what is needed to parse the trailers when creating `stream`, even though none of those are relevant to server-side uses of this type (this is what I had before)

Because of the behavior of not knowing when the trailers will be sent, I don't understand how I would make your suggestion work. If I were to put it in requestStream.

I'm going to take another stab at it without resorting to callbacks

Reply:
Now that I look at this more, the issue is specifically with responses that do not have a content-length. There are different behavior there, which is kind of annoying.